js/adapt-assessmentArticleModel.js

```diff
@@ -666,24 +671,24 @@ define([
             }
         
             this.set("_scoreAsPercent", scoreAsPercent);
-            this.set("_lastAttemptScoreAsPercent", scoreAsPercent)
+            this.set("_lastAttemptScoreAsPercent", scoreAsPercent);
 
             
             var questions = [];
-            for (var id in indexByIdQuestions) {
-                if (Adapt.findById(id).get("_isQuestionType")) {
-                    questions.push({
-                        _id: id,
-                        _isCorrect: indexByIdQuestions[id]
-                    });
+            for (id in indexByIdQuestions) {
+                if(indexByIdQuestions.hasOwnProperty(id)) {
+                    if (Adapt.findById(id).get("_isQuestionType")) {
```

Comment:
Doesn't need to stop you merging but I'd avoid the double nested if statements by concatanating the conditions with `&&`.